ping/rust/src/bin/testplan_0490.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use futures::StreamExt;
+use libp2pv0490::swarm::SwarmEvent;
+use libp2pv0490::*;
+use std::collections::HashSet;
+use std::time::Duration;
+use testplan::{run_ping, PingSwarm};
+
+#[async_std::main]
+async fn main() -> Result<()> {
+    let local_key = identity::Keypair::generate_ed25519();
+    let local_peer_id = PeerId::from(local_key.public());
+
+    let swarm = OrphanRuleWorkaround(Swarm::new(
+        development_transport(local_key).await?,
+        ping::Behaviour::new(
+            #[allow(deprecated)]
+            // TODO: Fixing this deprecation requires https://github.com/libp2p/rust-libp2p/pull/3055.
```

Comment:
we are not using the `derive` macro here, do we need `#[allow(deprecated)]` ?

Reply:
It is still a deprecated function!